### PR TITLE
fix: OpenTelemetry schema URL panic

### PR DESCRIPTION
**Pull request by disq**
`resource.Merge` returns an error only when schemas don't match.

### CODE REVIEW

File: serve/opentelemetry.go [hunk 1]
```diff
@@ -26,10 +26,15 @@ import (
 
 // newResource returns a resource describing this application.
 func newResource(p *plugin.Plugin) *resource.Resource {
+	def := resource.Default()
+	schemaURL := def.SchemaURL()
+	if schemaURL == "" {
+		schemaURL = semconv.SchemaURL
+	}
 	r, err := resource.Merge(
```
Comment (disq) [inline serve/opentelemetry.go:34]: This method always returns `r` (even with `err`, which the single value can be `ErrSchemaURLConflict`) so we could also do `if r != nil  { return r } panic(err)`

**Review by mnorbury — APPROVED** [review]

**Comment by erezrokah** [comment]
Is there a reproduction for the panic? Based on the panic stack:
```
panic: conflicting Schema URL: https://opentelemetry.io/schemas/1.26.0 and https://opentelemetry.io/schemas/1.24.0
goroutine 1 [running]:
github.com/cloudquery/plugin-sdk/v4/serve.newResource(0xc0006e92c0)
	/go/pkg/mod/github.com/cloudquery/plugin-sdk/v4@v4.38.2/serve/opentelemetry.go:21 +0x2e5
github.com/cloudquery/plugin-sdk/v4/serve.(*PluginServe).newCmdPluginServe.func1(0xc00063c908, {0xc0000fd600?, 0x4?, 0x14d0c66?})
	/go/pkg/mod/github.com/cloudquery/plugin-sdk/v4@v4.38.2/serve/plugin.go:153 +0x34e
github.com/spf13/cobra.(*Command).execute(0xc00063c908, {0xc0000fd550, 0xb, 0xb})
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc00063c608)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032
github.com/cloudquery/plugin-sdk/v4/serve.(*PluginServe).Serve(0xc0005d8540, {0x1726860, 0x220f220})
	/go/pkg/mod/github.com/cloudquery/plugin-sdk/v4@v4.38.2/serve/plugin.go:114 +0xee
main.main()
	/go/main.go:13 +0xaa
``` 

The issue is from an [old SDK version](https://github.com/cloudquery/plugin-sdk/pull/1635) (introduced in https://github.com/cloudquery/plugin-sdk/pull/1634/files) and fixed via https://github.com/cloudquery/plugin-sdk/pull/1605/files and https://github.com/cloudquery/plugin-sdk/pull/1607/files

**Comment by erezrokah** [comment]
OK something doesn't add up as `v4.38.2` was released after https://github.com/cloudquery/plugin-sdk/pull/1605

**Review by erezrokah — APPROVED** [review]
OK let's go with this fix though not sure how the mismatch can happen, probably related to the logic in https://github.com/open-telemetry/opentelemetry-go/blob/82c57cb8dd23ab537e60fbd1625bbb88b163f222/sdk/resource/resource.go#L230

**Pull request merged**